connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/ProtobufDataToCatalyst.scala

```diff
@@ -63,7 +64,16 @@ private[protobuf] case class ProtobufDataToCatalyst(
   @transient private lazy val fieldsNumbers =
     messageDescriptor.getFields.asScala.map(f => f.getNumber).toSet
 
-  @transient private lazy val deserializer = new ProtobufDeserializer(messageDescriptor, dataType)
+  @transient private lazy val deserializer = {
+    val typeRegistry = descFilePath match {
+      case Some(path) if protobufOptions.convertAnyFieldsToJson =>
+        ProtobufUtils.buildTypeRegistry(path) // This loads all the messages in the file.
+      case None if protobufOptions.convertAnyFieldsToJson =>
+        ProtobufUtils.buildTypeRegistry(messageDescriptor) // Loads only connected messages.
+      case _ => TypeRegistry.getEmptyTypeRegistry // Default. Json conversion is not enabled.
+    }
+    new ProtobufDeserializer(messageDescriptor, dataType, typeRegistry = typeRegistry)
+  }
```

Comment:
@rangadi Is it possible to replace `buildDescriptor(descriptor, messageName)` with `ProtobufUtils.buildTypeRegistry(path)` ? and use `typeRegistry.find(messageName)`